### PR TITLE
[Bug] Null reference in BitcoinSerializableJsonConverter

### PR DESCRIPTION
**Pull request by fassadlr**
Raised by PistolPete in discord

### CODE REVIEW

File: src/NBitcoin/IBitcoinSerializable.cs [hunk 1]
```diff
@@ -42,20 +42,28 @@ public static void ReadWrite(this IBitcoinSerializable serializable, byte[] byte
             ReadWrite(serializable, new MemoryStream(bytes), false, consensusFactory, version);
         }
 
+        private static BitcoinStream CreateBitcoinStream()
+        {
+            return new BitcoinStream(Stream.Null, true)
+            {
+                ConsensusFactory = new DefaultConsensusFactory(),
```
Comment (dangershony) [inline src/NBitcoin/IBitcoinSerializable.cs]: Possibly I would set that in the constructor

Comment (bokobza) [reply src/NBitcoin/IBitcoinSerializable.cs]: ah, just raised an issue with that problem.

**Comment by bokobza** [comment]
@fassadlr is everything tested and syncing smoothly?

File: src/NBitcoin/IBitcoinSerializable.cs (hunk 1 above)
Comment (fassadlr) [reply src/NBitcoin/IBitcoinSerializable.cs]: @dangershony can we look at maybe doing that across the board after this fix has gone in?

**Comment by fassadlr** [comment]
@bokobza I have synced stratis testnet and tested with PistolPetes wallet to ensure that the node starts up correctly. Staking also works as per usual. I'm busy syncing the other chainz.

File: src/NBitcoin/IBitcoinSerializable.cs (hunk 1 above)
Comment (bokobza) [reply src/NBitcoin/IBitcoinSerializable.cs]: I think he just means that yo udon't need to have a new static method to instantiate the BitcoinStream object. Just bring the `new BitcoinStream(...` inline to from where it's called.

Comment (bokobza) [reply src/NBitcoin/IBitcoinSerializable.cs]: ah no he's saying that you should set up the `DefaultConsensusFactory` in the ctor.
What I'm saying is that you don't need an extra method :-)

**Comment by bokobza** [comment]
Also look at this issue I raised yesterday, it's related.
https://github.com/stratisproject/StratisBitcoinFullNode/issues/1582

This happens when you hav a wallet in your folder, so that's something to be tested as well.

**Comment by bokobza** [comment]
>  tested with PistolPete's wallet 

Cool, missed that.
so just the CreateBitcoinStream method then.

**Comment by Aprogiena** [comment]
I missing some kind of explanation of the bug and the fix. Where does DefaultConsensusFactory come from and how is that we don't need to get it from the Network?

**Pull request merged**